pkg/gatewayserver/gatewayserver.go

```diff
@@ -384,7 +384,7 @@ var (
 // This method should only be used for request contexts.
 func (gs *GatewayServer) FillGatewayContext(ctx context.Context, ids ttnpb.GatewayIdentifiers) (context.Context, ttnpb.GatewayIdentifiers, error) {
 	ctx = gs.FillContext(ctx)
-	if ids.IsZero() {
+	if ids.IsZero() || ids.EUI.IsZero() {
```

Comment:
`ids.EUI` is nillable, but since `ids.IsZero` is false, `ids.GatewayId != "" || ids.EUI != nil`. So you may `panic`.

Reply:
Yeah saw that and fixed it..:)

Reply:
Yeah saw that and fixed it..:)